src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptionsConfigExtensions.cs

```diff
@@ -78,7 +84,14 @@ public static class MessageDrivenSubscriptionsConfigExtensions
             Guard.AgainstNull(nameof(eventNamespace), eventNamespace);
             Guard.AgainstNullAndEmpty(nameof(publisherEndpoint), publisherEndpoint);
 
+            if (publisherEndpoint.Contains("@"))
+                throw new ArgumentException(string.Format(endpointNameExceptionMessageTemplate, publisherEndpoint));
+
             routingSettings.Settings.GetOrCreate<Publishers>().Add(eventAssembly, eventNamespace, publisherEndpoint);
         }
+
+        static string endpointNameExceptionMessageTemplate = "A logical endpoint name should not contain '@', but received '{0}'. "
+                                          + "To specify an endpoint's address use the instance mapping file for MSMQ transport or refer to the routing documentation.";
```

Comment:
Grammar nitpick:

"To specify an endpoint's address, use the instance mapping file for the MSMQ transport, or refer to the routing documentation."